tool/src/main/java/org/wildfly/security/tool/Command.java

```diff
@@ -270,4 +326,177 @@ protected Supplier<Provider[]> getProvidersSupplier(final String providersList)
             }
         };
     }
+
+    /**
+     * Acquire a given secret key from a {@link CredentialStore}.
+     *
+     * @param alias the name for a secret key within the CredentialStore
+     * @return the requested {@link SecretKey}, or {@code null} if it could not be retrieved
+     */
+    SecretKey getSecretKey(Boolean createCredentialStore, String credentialStoreLocation, String alias, Boolean populate,
+                           String inputRealmLocation, int descriptorBlockCount) throws Exception {
```

Comment:
Does this throw anything other than `CredentialStoreException`? Can we list the specific exceptions?

Reply:
Have the parameters been validated before this helper method is called?

Reply:
> Does this throw anything other than `CredentialStoreException`? Can we list the specific exceptions?

Sure, the other exception is `NoSuchAlgorithmException` from getting a cred store instance

> Have the parameters been validated before this helper method is called?

They should at least be non-null, but looks like the credential store isn't checked to see if the path exists.

Reply:
If we've already checked for non-null before calling this helper method, that's fine. I don't think we need to validate the path exists at this point. I believe that will be checked when calling credentialStore.initialize below.

Reply:
alright, I'll leave it as is